A few devnet related fixes

codablock opened the pull request:
These fixes are originally from the DIP3 branch but should be included into develop ASAP, as otherwise devnet users will have issues connecting to a devnet. See individual commits.

UdjinM6 commented on src/net.cpp:
```diff
@@ -1990,10 +1995,14 @@ bool CConnman::OpenNetworkConnection(const CAddress& addrConnect, bool fCountFai
+        if (IsBanned(addrConnect) ||
             FindNode(addrConnect.ToStringIPPort()))
             return false;
+        if (IsLocal(addrConnect) && (!Params().AllowMultiplePorts() || addrConnect.GetPort() == GetListenPort()))
```
Same here
```
bool fAllowLocal = Params().AllowMultiplePorts() && addrConnect.GetPort() != GetListenPort();
if ((!fAllowLocal && IsLocal(addrConnect)) ||
    (!Params().AllowMultiplePorts() && FindNode((CNetAddr)addrConnect)) ||
    (Params().AllowMultiplePorts() && FindNode((CService)addrConnect)) ||
    IsBanned(addrConnect) ||
    FindNode(addrConnect.ToStringIPPort()))
    return false;
```

thephez commented:
As discussed on Slack (and more related), non-standard ports seem to cause re-connection issues. If a node starts on non-default ports, it will not connect automatically to a remote peer running on that port unless `addnode` is used every time the node starts. It _will_ connect to a remote peer that is running on a non-standard port if it is running on the default.

To summarize:

Local Port | Remote Port| Connection
-|-|-
18999 | 18999 | No automatic re-connect (after initial `addnode`)
**19999** | 18999 | Automatic re-connect works (after initial `addnode`)

Also, a little off-topic for the PR probably, but why does `-devnet` require assigning `-port` and `-rpcport` when there are defaults set in [chainparams](https://github.com/dashpay/dash/blob/master/src/chainparams.cpp#L472)? :thinking:

codablock commented:
Added suggested changes. I however left the multiple ifs in OpenNetworkConnection intact, as I always found the original if extremely hard to read. I added a few comments to make it easier to read.

UdjinM6 commented:
Tested and it seems to work as expected.

utACK -> ACK
